### PR TITLE
fix(health): harden registry with overall timeout and defensive slice copy

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- Replace array-based result collection in `runChecks` with a channel-based approach that respects parent context cancellation, preventing indefinite HTTP handler hangs when a health check ignores its per-check context timeout
- Add `slices.Clone` on `MultiResultCheck.RunMulti` return values to prevent the orchestrator from mutating a caller's shared/cached slice
- Modernize goroutine launch to `sync.WaitGroup.Go` (Go 1.25+) and unify slice copy idioms to `slices.Clone`

**Root cause (timeout):** `runChecks` spawned goroutines with 10s per-check contexts, then called `wg.Wait()` unconditionally. If a check ignored its context (e.g., `gopsutil.SensorsTemperatures` or `disk.Usage` syscalls), `wg.Wait()` blocked forever, hanging the `/api/v2/system/diagnostics/run` endpoint.

**Root cause (slice mutation):** After `RunMulti` returned `[]Result`, the loop mutated `DurationMS` and `Timestamp` in place. If any `RunMulti` implementation returned a cached slice, this would be a data race.

**Design:** After context expiration, a 100ms grace period lets context-aware checks finish and report their own status before the orchestrator synthesizes `StatusUnknown` for unfinished checks. Results are collected via a buffered channel (never closed, so hung goroutines can safely send later without panicking).

## Test Plan

- [x] `TestRunChecks_OverallTimeout`: fast check returns healthy, hung check (ignores context) gets synthetic `StatusUnknown`
- [x] `TestRunChecks_OverallTimeout_MultiResultNilSlice`: multi-check returning nil + hung check correctly distinguishes "completed with zero results" from "did not complete"
- [x] `TestRunChecks_MultiResultDefensiveCopy`: verifies original slice is not mutated by orchestrator
- [x] All 63 existing tests pass with `-race` flag
- [x] Zero lint issues

## Summary by CodeRabbit

* **Bug Fixes**
  * Health checks now return a clear "check did not complete within deadline" status when they exceed the overall deadline.
  * Orchestration reliably collects and orders results even when some checks hang or return multiple results.
  * Multi-check results are handled defensively to avoid unexpected mutations or missing entries.

* **Tests**
  * Added coverage for timeout behavior, hanging checks, and multi-result edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/tphakala/birdnet-go/pull/3251?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: internal/health/registry.go [hunk 1]
```diff
@@ -116,15 +130,54 @@ func runChecks(ctx context.Context, checks []Check) []Result {
...
+	case <-ctx.Done():
+		// Give context-aware checks a brief grace period to finish and
+		// report their own status before we synthesize StatusUnknown.
+		grace := time.NewTimer(contextGracePeriod)
+		select {
+		case <-done:
+			grace.Stop()
+		case <-grace.C:
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/health/registry.go]: Fixed in 8e05128 - replaced the waiter goroutine + WaitGroup with count-based collection from resCh directly. The main goroutine now reads from the buffered channel until all checks report or the grace timer fires, eliminating both the leaked waiter goroutine and the drain loop.

File: internal/health/registry.go [hunk 2]
```diff
@@ -116,15 +129,50 @@ func runChecks(ctx context.Context, checks []Check) []Result {
...
+				case cr := <-resCh:
+					completed[cr.idx] = cr.results
+					received++
+				case <-grace.C:
+					received = len(checks)
+				}
+			}
+			grace.Stop()
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/health/registry.go]: Fixed in d3dad7c - switched to [][]Result slice with []bool tracking and labeled break.

File: internal/health/registry.go [hunk 3]
```diff
@@ -116,15 +129,50 @@ func runChecks(ctx context.Context, checks []Check) []Result {
...
 
 	results := make([]Result, 0, len(checks))
-	for _, o := range outputs {
-		results = append(results, o.results...)
+	for i, c := range checks {
+		if rs, ok := completed[i]; ok {
+			results = append(results, rs...)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/health/registry.go]: Done in the same commit.
